src/nvme/types.h

```diff
@@ -3319,6 +3319,92 @@ struct nvme_media_unit_stat_log {
 	struct nvme_media_unit_stat_desc mus_desc[];
 };
 
+/**
+ * struct nvme_media_unit_config_desc 
+ * @muid: Media Unit Identifier
+ * @mudl: Media Unit Descriptor Length
+ * 
+ * Media Unit Configuration Descriptor
+ * Structure Definitions
+ */
+struct nvme_media_unit_config_desc {
+	__le16	muid;
+	__u8	rsvd2[4];
+	__le16	mudl;
+};
+
+/**
+ * struct nvme_channel_config_desc -
+ * @chanid:	Channel Identifier
+ * @chmus:	Number Channel Media Units
+ * 
+ * Channel Configuration Descriptor
+ * Structure Definitions
+ */
+struct nvme_channel_config_desc {
+	__le16	chanid;
+	__le16	chmus;
+	struct nvme_media_unit_config_desc mu_config_desc[];
+};
+
+/**
+ * struct nvme_end_grp_config_desc -
+ * @endgid:		Endurance Group Identifier
+ * @cap_adj_factor:	Capacity Adjustment Factor
+ * @tegcap:		Total Endurance Group Capacity
+ * @segcap:		Spare Endurance Group Capacity
+ * @end_est:		Endurance Estimate
+ * @egsets:		Number of NVM Sets
+ * @nvmsetid:		NVM Set Identifier
+ * @egchans:		Number of Channels
+ * 
+ * Endurance Group Configuration Descriptor
+ * Structure Definitions
+ */
+struct nvme_end_grp_config_desc {
+	__le16	endgid;
+	__le16	cap_adj_factor;
+	__u8	rsvd4[12];
+	__u8	tegcap[16];
+	__u8	segcap[16];
+	__u8	end_est[16];
+	__u8	rsvd64[16];
+	__le16	egsets;
+	__le16	nvmsetid[256];
+	__le16	egchans;
+	struct nvme_channel_config_desc chan_config_desc[];
+};
+
+/**
+ * struct nvme_cap_config_desc -
+ * @cap_config_id:	Capacity Configuration Identifier
+ * @domainid:		Domain Identifier
+ * @egcn:		Number Endurance Group Configuration
+ * 			Descriptors
+ * 
+ * Capacity Configuration structure definitions
+ */
+struct nvme_capacity_config_desc {
```

Comment:
This is even missing in the spec I am looking at. Looks like I am reading the wrong spec. Can you point me to right one?

Reply:
You could find the Capacity Configuration Descriptor in Figure 251(Page-235) in NVMe 2.0 spec

Reply:
Ah yes. I got confused by Figure 253 (Channel Configuration Descriptor) which missing from this patch. Could you add it to make this section complete?

Reply:
Channel configuration descriptor is already part of this patch.
struct nvme_channel_config_desc {
	__le16	chanid;
	__le16	chmus;
	struct nvme_media_unit_config_desc mu_config_desc[];
};
Could you please check

Reply:
Right you are! Sorry for the noise.